### PR TITLE
remove mutating webhook from Seldon Core Operator

### PR DESCRIPTION
**Pull request by RafalSkolasinski**
**What this PR does / why we need it**:

This removes the mutating webhook: `Mutatingwebhookconfigurations` is not installed into the cluster.
From now on, the `SeldonDeployment` resource written to `etcd` of k8s will be exactly the same as resource applied by the user. 

This will simplify GitOps operations as the parity between applied and visible resources will be present. 
To see extra information that was written into CR before one can take a look at the generated deployments.
They will contain all information about the assigned ports, init containers, etc.

**Which issue(s) this PR fixes**:

Closes https://github.com/SeldonIO/seldon-core/issues/2817
Closes https://github.com/SeldonIO/seldon-core/issues/2811

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
Mutating Webhook is removed. `SeldonDeployment` Resource is written to `etcd` without any alterations.
```

### CODE REVIEW

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **rafalskolasinski**
You can assign the PR to them by writing `/assign @rafalskolasinski` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by seldondev** [comment]
Mon Jan 18 18:53:55 UTC 2021
The logs for [lint] [2] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/2.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=2

**Comment by seldondev** [comment]
Mon Jan 18 18:54:11 UTC 2021
The logs for [pr-build] [1] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/1.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=1

**Comment by seldondev** [comment]
Mon Jan 18 18:57:03 UTC 2021
The logs for [pr-build] [3] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/3.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=3

**Comment by seldondev** [comment]
Mon Jan 18 18:57:16 UTC 2021
The logs for [lint] [4] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/4.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=4

**Comment by seldondev** [comment]
Mon Jan 18 18:59:25 UTC 2021
The logs for [pr-build] [5] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/5.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=5

**Comment by seldondev** [comment]
Mon Jan 18 18:59:32 UTC 2021
The logs for [lint] [6] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/6.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=6

**Comment by seldondev** [comment]
Tue Jan 19 10:38:43 UTC 2021
The logs for [lint] [8] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/8.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=8

**Comment by seldondev** [comment]
Tue Jan 19 10:38:42 UTC 2021
The logs for [integration] [10] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/10.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=10

**Comment by seldondev** [comment]
Tue Jan 19 10:38:40 UTC 2021
The logs for [pr-build] [7] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/7.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=7

**Comment by seldondev** [comment]
Tue Jan 19 10:38:48 UTC 2021
The logs for [notebooks] [9] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/9.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=9

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go
Comment (axsaucedo) [inline operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:543]: All the code related to the webhook should be removed as well, as there are several functions in this file which would no longer be called

**Review by axsaucedo — CHANGES_REQUESTED** [review]

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go
Comment (cliveseldon) [reply operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:543]: We need to keep the mutating functionality which is called from Reconcile

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | ff2f6fd36b2b7f8f183ec7117ce181979ad35d1d | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Seems integration tests are not going well
```
23 failed, 12 passed
```
gonna have to look locally into them

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go
Comment (axsaucedo) [reply operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:543]: Ok makes sense, although it woudl still be worth moving it from the webhook file as it would no longer be used by the webhook itself

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | ff2f6fd36b2b7f8f183ec7117ce181979ad35d1d | [link]() | `/test integration`
notebooks | ff2f6fd36b2b7f8f183ec7117ce181979ad35d1d | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
[APPROVALNOTIFIER] This PR is **NOT APPROVED**

This pull-request has been approved by:
To complete the [pull request process](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process), please assign **axsaucedo**
You can assign the PR to them by writing `/assign @axsaucedo` in a comment when ready.

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=SeldonIO%2Fseldon-core).

<details open>
Needs approval from an approver in each of these files:

- **[OWNERS](https://github.com/SeldonIO/seldon-core/blob/master/OWNERS)**

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

**Comment by RafalSkolasinski** [comment]
Let's hope it is fixed now.
/test integration

**Comment by seldondev** [comment]
Tue Jan 19 12:04:01 UTC 2021
The logs for [lint] [12] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/12.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=12

**Comment by seldondev** [comment]
Tue Jan 19 12:04:05 UTC 2021
The logs for [pr-build] [11] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/11.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=11

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go
Comment (RafalSkolasinski) [reply operator/apis/machinelearning.seldon.io/v1/seldondeployment_webhook.go:543]: Actually the code will be required in the webhooks -> validating webhooks were receiving defaulted resource before.
Now they need to apply defaulting on their own before validating -> I suspect that was  the reason why tests failed in previous run.

**Comment by seldondev** [comment]
Tue Jan 19 12:06:38 UTC 2021
The logs for [integration] [13] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/13.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=13

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | ff2f6fd36b2b7f8f183ec7117ce181979ad35d1d | [link]() | `/test notebooks`
integration | 055cc0f87feb146948f60539bdc7661be1487182 | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Only 13 failures now...

**Comment by seldondev** [comment]
Tue Jan 19 15:47:26 UTC 2021
The logs for [pr-build] [14] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/14.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=14

**Comment by seldondev** [comment]
Tue Jan 19 15:47:40 UTC 2021
The logs for [integration] [16] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/16.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=16

**Comment by seldondev** [comment]
Tue Jan 19 15:47:43 UTC 2021
The logs for [lint] [15] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/15.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=15

**Comment by seldondev** [comment]
Tue Jan 19 18:01:06 UTC 2021
The logs for [notebooks] [17] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/17.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=17

**Comment by RafalSkolasinski** [comment]
All tests passed!!!!

**Comment by RafalSkolasinski** [comment]
I believe this is ready to get merged @cliveseldon @axsaucedo

**Comment by seldondev** [comment]
Thu Jan 21 13:06:07 UTC 2021
The logs for [lint] [19] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/19.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=19

**Comment by seldondev** [comment]
Thu Jan 21 13:06:15 UTC 2021
The logs for [pr-build] [18] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/18.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=18

**Comment by seldondev** [comment]
Thu Jan 21 13:07:17 UTC 2021
The logs for [integration] [21] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/21.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=21

**Comment by seldondev** [comment]
Thu Jan 21 13:07:19 UTC 2021
The logs for [notebooks] [20] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/20.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=20

**Comment by RafalSkolasinski** [comment]
Moved defaulting code outside of webhook file.

/test integration
/test notebooks

**Comment by seldondev** [comment]
Thu Jan 21 13:33:59 UTC 2021
The logs for [integration] [25] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/25.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=25

**Comment by seldondev** [comment]
Thu Jan 21 13:34:02 UTC 2021
The logs for [pr-build] [22] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/22.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=22

**Comment by seldondev** [comment]
Thu Jan 21 13:34:36 UTC 2021
The logs for [lint] [23] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/23.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=23

**Comment by seldondev** [comment]
Thu Jan 21 13:34:32 UTC 2021
The logs for [notebooks] [24] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/24.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=24

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | d187a0874313f9a27d5191e05aafce2af71f6150 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Single notebook failure in: TestNotebooks.test_istio_examples
Let's see what run scheduled later will show

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
notebooks | 43184adab444322db78b8344a98cc4fbf922e142 | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Single notebook failure in: TestNotebooks.test_custom_metrics
Seems that one is still flaky...
/test notebooks

**Comment by seldondev** [comment]
Thu Jan 21 15:33:34 UTC 2021
The logs for [notebooks] [26] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/26.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=26

**Comment by cliveseldon** [comment]
We need to remove `operator/utils/k8s/webhook::CreateMutatingWebhookConfigurationFromFile` which is used when the operator is run in the mode to create resources. This is used in Openshift where webhooks need  to be created by the manager. You can test by setting in helm `--set managerCreateResources=true`

File: operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go
Comment (cliveseldon) [inline operator/apis/machinelearning.seldon.io/v1/seldondeployment_types.go:507]: Can we move to separate file rather than add. maybe `defaults.go` ?

**Review by cliveseldon — CHANGES_REQUESTED** [review]

**Comment by seldondev** [comment]
Fri Jan 22 12:09:57 UTC 2021
The logs for [pr-build] [27] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/27.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=27

**Comment by seldondev** [comment]
Fri Jan 22 12:10:09 UTC 2021
The logs for [notebooks] [29] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/29.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=29

**Comment by seldondev** [comment]
Fri Jan 22 12:10:12 UTC 2021
The logs for [lint] [28] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/28.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=28

**Comment by seldondev** [comment]
Fri Jan 22 12:10:19 UTC 2021
The logs for [integration] [30] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/30.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=30

**Comment by RafalSkolasinski** [comment]
had to rebase on master
/test integration
/test notebooks

**Comment by seldondev** [comment]
Fri Jan 22 14:10:26 UTC 2021
The logs for [integration] [34] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/34.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=34

**Comment by seldondev** [comment]
Fri Jan 22 14:10:27 UTC 2021
The logs for [lint] [32] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/32.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=32

**Comment by seldondev** [comment]
Fri Jan 22 14:10:35 UTC 2021
The logs for [notebooks] [33] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/33.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=33

**Comment by seldondev** [comment]
Fri Jan 22 14:10:37 UTC 2021
The logs for [pr-build] [31] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/31.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=31

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Fri Jan 22 18:14:53 UTC 2021
The logs for [integration] [35] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/35.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=35

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
something had to go wrong during rebase, gonna fix on Monday

**Comment by seldondev** [comment]
Fri Jan 22 20:42:46 UTC 2021
The logs for [integration] [36] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/36.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=36

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Sat Jan 23 17:14:22 UTC 2021
The logs for [integration] [37] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/37.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=37

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Mon Jan 25 09:41:04 UTC 2021
The logs for [integration] [38] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/38.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=38

**Comment by seldondev** [comment]
@RafalSkolasinski: The following test **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
currently blocked by https://github.com/SeldonIO/seldon-core/pull/2886, need to be rebased once https://github.com/SeldonIO/seldon-core/pull/2886 is merged

**Comment by seldondev** [comment]
Tue Jan 26 14:01:07 UTC 2021
The logs for [lint] [40] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/40.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=40

**Comment by seldondev** [comment]
Tue Jan 26 14:01:22 UTC 2021
The logs for [pr-build] [39] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/39.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=39

**Comment by seldondev** [comment]
Tue Jan 26 14:02:48 UTC 2021
The logs for [integration] [42] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/42.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=42

**Comment by seldondev** [comment]
Tue Jan 26 14:02:59 UTC 2021
The logs for [notebooks] [41] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/41.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=41

**Comment by RafalSkolasinski** [comment]
Rebased onto master, nicely squashed commit and also tested once again locally that mutatingwebhook is not getting created. 
Running tests in CI - should be good to merge after.

@cliveseldon, don't know if you want to have a final look over the changes?

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`
notebooks | 1e43cc822ab8dd858b26e9ba4237e79432b47f2f | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by RafalSkolasinski** [comment]
Hmm... notebook tests failed on installation of istio
```
✘ Addons encountered an error: failed to wait for resource: resources not ready after 5m0s: timed out waiting for the condition
Deployment/istio-system/grafana
- Pruning removed resourcesError: failed to apply manifests: errors occurred during operation
make: *** [Makefile:107: install_istio] Error 1
Kind setup returned errors
```
/retest

**Comment by seldondev** [comment]
Tue Jan 26 15:08:49 UTC 2021
The logs for [notebooks] [43] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/43.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=43

**Comment by seldondev** [comment]
@RafalSkolasinski: The following tests **failed**, say `/retest` to rerun them all:

Test name | Commit | Details | Rerun command
--- | --- | --- | ---
integration | 2888070e3caa728600883b655e3a326a0278fe7b | [link]() | `/test integration`
notebooks | 1e43cc822ab8dd858b26e9ba4237e79432b47f2f | [link]() | `/test notebooks`



<details>

Instructions for interacting with me using PR comments are available [here](https://git.k8s.io/community/contributors/guide/pull-requests.md).  If you have questions or suggestions related to my behavior, please file an issue against the [jenkins-x/lighthouse](https://github.com/jenkins-x/lighthouse/issues/new?title=Command%20issue:) repository. I understand the commands that are listed [here](https://go.k8s.io/bot-commands).
</details>

**Comment by seldondev** [comment]
Tue Jan 26 16:41:28 UTC 2021
The logs for [notebooks] [44] will show after the pipeline context has finished.  
https://github.com/SeldonIO/seldon-core/blob/gh-pages/jenkins-x/logs/SeldonIO/seldon-core/PR-2852/44.log

impatient try  
jx get build logs SeldonIO/seldon-core/PR-2852 --build=44

**Comment by RafalSkolasinski** [comment]
All testes passed!!! Let's land this :rocket:

**Comment by axsaucedo** [comment]
Nice one

**Pull request merged**
